book/src/user_manual/persistence.md

```diff
@@ -4,13 +4,12 @@ The state of a given `MlsGroup` instance is continuously written to the configur
 `StorageProvider`. Later, the `MlsGroup` can be loaded from the provider using
 the `load` constructor, which can be called with the respective storage provider
 as well as the `GroupId` of the group to be loaded. For this to work, the group
-must be stored in the provider.
+must have been written to the provider previously.
 
 ## Forward-Secrecy Considerations
 
-The persisted `MlsGroup` state contains
-private key material. As a consequence, the application needs to delete old group
-states to achieve Forward-Secrecy w.r.t. that key material. Since, as detailed
-above, an old group state is stale immediately after most group operations,
-we recommend that the `StorageProvider` deletes old group state when values are
-overwritten.
+The persisted `MlsGroup` state contains private key material. As a consequence,
+delete old group states need to be deleted in order to achieve Forward-Secrecy
+w.r.t. that key material. Since, as detailed above, an old group state is stale
+immediately after most group operations, we recommend that the `StorageProvider`
+deletes old group state when values are overwritten.
```

Comment:
I'm still confused as to why we have to recommend this. Doesn't OpenMLS delete the values that are no longer needed using the trait?

Reply:
> I'm still confused as to why we have to recommend this. Doesn't OpenMLS delete the values that are no longer needed using the trait?

OpenMLS tells the provider to delete or overwrite the data, yes. This section tells implementors of providers that these write-requests should be applied quickly and irreversibly. For example, it doesn't make much sense to implement the storage provider on top of git or some append-only copy-on-write data structure, because then we'd keep around old keys.